packages/dtslint-runner/src/prepareAllPackages.ts

```diff
@@ -1,27 +1,54 @@
-import { AllPackages, getDefinitelyTyped, parseDefinitions } from "@definitelytyped/definitions-parser";
-import { joinPaths, loggerWithErrors } from "@definitelytyped/utils";
+import { getDefinitelyTyped, parseDefinitions, PreparePackagesResult } from "@definitelytyped/definitions-parser";
+import { execAndThrowErrors, loggerWithErrors, pnpmInstallFlags, sleep } from "@definitelytyped/utils";
 import { checkParseResults } from "./check-parse-results";
-import { installDependencies } from "./prepareAffectedPackages";
-import { PreparePackagesOptions, PreparePackagesResult } from "./types";
 
-export async function prepareAllPackages({
-  definitelyTypedPath,
-  noInstall,
-  nProcesses,
-}: PreparePackagesOptions): Promise<PreparePackagesResult> {
-  const typesPath = joinPaths(definitelyTypedPath, "types");
+export async function prepareAllPackages(
+  definitelyTypedPath: string,
+  clone: boolean,
+  nProcesses: number
+): Promise<PreparePackagesResult> {
   const [log] = loggerWithErrors();
   const options = {
     definitelyTypedPath,
     progress: false,
     parseInParallel: nProcesses > 1,
   };
   const dt = await getDefinitelyTyped(options, log);
-  await parseDefinitions(dt, nProcesses ? { definitelyTypedPath, nProcesses } : undefined, log);
-  await checkParseResults(/*includeNpmChecks*/ false, dt);
-  const allPackages = await AllPackages.read(dt);
-  if (!noInstall) {
-    await installDependencies(allPackages.allTypings(), typesPath);
+  const allPackages = await parseDefinitions(dt, nProcesses ? { definitelyTypedPath, nProcesses } : undefined, log);
+  if (clone) {
+    await installAllDependencies(definitelyTypedPath);
+  }
+  const errors = checkParseResults(allPackages);
+  if (errors.length) {
+    throw new Error(errors.join("\n"));
+  }
+  return {
+    packageNames: new Set(allPackages.allTypings().map(({ subDirectoryPath }) => subDirectoryPath)),
+    dependents: new Set(),
+  };
+}
+const npmRetryCount = 5;
+export async function installAllDependencies(typesPath: string): Promise<void> {
+  console.log("Installing NPM dependencies...");
+  const cmd = `pnpm install ${pnpmInstallFlags}`;
+  console.log(`  ${typesPath}: ${cmd}`);
+  let lastError;
+  for (let i = 0; i < npmRetryCount; i++) {
+    try {
+      const stdout = await execAndThrowErrors(cmd, typesPath);
```

Comment:
Is this execing at the types root, not the DT root? We probably want to run at the repo root, though maybe it's working because there's no package.json in that dir?